Группировка пользователей по возрасту

elka1907 opened the pull request:
Привет! 
Прошу проверить.
Спасибо

elka1907 committed:
Группировка пользователей по возрасту

MarselYsup commented on src/main/java/school/faang/User.java:
```diff
@@ -0,0 +1,30 @@
+
+import lombok.Data;
+import lombok.AllArgsConstructor;
+import java.util.*;
```
Не стоит использовать импорт всего (*). Делайте только точечные импорты, необходимые для текущего класса. Это улучшает читаемость кода и упрощает рефакторинг.

Если у тебя IDEA прочитай статью https://www.baeldung.com/intellij-disable-wildcard-import
тут написано, как можно эту вещь отключить

MarselYsup commented on src/main/java/school/faang/User.java:
```diff
@@ -0,0 +1,30 @@
+import lombok.AllArgsConstructor;
+import java.util.*;
+
+@Data
```
не используй @Data - если тебе не нужны все его аннотации, так например он генерирует ```@RequiredArgsConstructor``` 
лучше просто сделай связку 
```
@Getter
@ToString
@AllArgsConstructor
@EqualsAndHashcode
```

MarselYsup commented on src/main/java/school/faang/User.java:
```diff
@@ -0,0 +1,30 @@
+public class User {
+    private String name;
+    private int age;
+    private String placeOfWork;
```
в задании поле требуется назвать ```workplace```

MarselYsup commented on src/main/java/school/faang/User.java:
```diff
@@ -0,0 +1,30 @@
+package school.faang;
```
советую делать для каждой задачи свой отдельный подпакет, так задачи не будут складываться в одну кучу

elka1907 replied:
не поняла что требуется сделать. У меня была папка school.faang в ней Main and User. 
Что требуется сделать?

MarselYsup replied:
у тебя будет еще много задач и если добавлять их в один подпакет, будет сложно потом разобраться к какой задачи написаны классы
просто сделай отдельную подпапку в 
```
school.faang.groupusers
```
и тд в других задачах можно будет добавить отдельный подпакет

```
school.faang.recursive
school.faang.library
и тд
```

MarselYsup commented on src/main/java/school/faang/Main.java:
```diff
@@ -1,4 +1,45 @@
 package school.faang;
 
+import java.util.*;
```
аналогично "Не стоит использовать импорт всего (*)."

elka1907 committed:
Корректировки:
1.main: import;
2. user: add import lombok, correct import, correct @DaTa, correct String workplace

elka1907 committed:
Корректировки:
1.main: import;
2. user: add import lombok, correct import, correct @DaTa, correct String workplace

elka1907 committed:
Корректировки:
1.main: import;
2. user: add import lombok, correct import, correct @DaTa, correct String workplace

elka1907 commented:
merge conflicts как решить конфликт? Спасибо

elka1907 commented:
merge conflicts как решить конфликт? спасибо

MarselYsup commented:
Привет! Обратись в чатик /help, там тебе помогут решить конфликт с мром

elka1907 commented:
Сделала следующее (конфликтов не было, видимо я все это уже удалила):
Способ 2: Локально через IDE (рекомендуется для сложных конфликтов): 
1 Открой терминал и перейди в свою ветку: git checkout твоя-ветка
2 Подтяни изменения из целевой ветки: git pull origin целевая-ветка
3  выполни: git add .
4 Затем: git commit -m "Resolve merge conflicts"
5 И наконец: git push origin твоя-ветка
PR не обновился.
Как еще решить проблему?
Спасибо
<img width="815" height="155" alt="33" src="https://github.com/user-attachments/assets/ceb7d529-cf5f-40b2-a11f-920922b9d58d" />

MarselYsup commented:
Проблема в том, что твоя основная ветка elka1907 ушла по коммитам дальше от твоей ветки  elka1907-BJS2-85655 из задачи, для этого нужно локально решить конфликт между этими ветками, лучше используй UI idea-и и с помощью него реши конфликт, тебе нужно прям кодом решить, что остается при слиянии веток  - вообще конфликт возник как раз из-за того что ты делал без подпакетов классы с одинаковым названием 
ты обратилась в чатик /help?

elka1907 committed:
corrections:
1. comment for PR

elka1907 commented:
вроде как решила проблему.
